client/src/store.js

```diff
@@ -20,10 +20,31 @@ export default new Vuex.Store({
   },
   actions: {
     async fetchPipelines({ commit, state }) {
-      if (state.pipelines.length === 0) {
+      if (Object.keys(state.pipelines).length === 0) {
```

Comment:
This raises a code smell for me, mainly because in my opinion, if we call an action it's because we want to fetch _or_ refresh data. With this check, if the pipelines have been fetched once, we won't be able to get a new list until the application is refreshed. I don't think there's a current use case for updating the pipeline list, but in the future there might be, specifically related to the pipeline training feature.

Reply:
This behavior isn't being changed in this PR.   Pipelines will currently not change without a full redeploy, so there's no reason to fetch it repeatedly.  If they start to change based on some other criteria, we can change to code to detect and re-fetch.

Reply:
Removal of that line would have it only refresh on navigation away from /home, to either jobs or an annotation.  If we want it to refresh everytime there would be little benefit to putting it in a vuex store (what else uses the `state.pipelines`?)

Reply:
Good point, I don't actually see why it's in the store, since it looks like only `RunPipelineMenu.vue` uses it.

Reply:
It's in the store because `RunPipelineMenu.vue` can be used in more than one place in the app.  I'm about to put one into Viewer.vue.   Avoids an API call on every instantiation of the component.